src/box/box.cc

```diff
@@ -3539,23 +3550,44 @@ bootstrap(const struct tt_uuid *instance_uuid,
 	 * with connecting to 'replication_connect_quorum' masters.
 	 * If this also fails, throw an error.
 	 */
-	box_restart_replication();
-
-	struct replica *master = replicaset_find_join_master();
-	assert(master == NULL || master->applier != NULL);
-
-	if (master != NULL && !tt_uuid_is_equal(&master->uuid, &INSTANCE_UUID)) {
-		bootstrap_from_master(master);
-		/* Check replica set UUID */
-		if (!tt_uuid_is_nil(replicaset_uuid) &&
-		    !tt_uuid_is_equal(replicaset_uuid, &REPLICASET_UUID)) {
-			tnt_raise(ClientError, ER_REPLICASET_UUID_MISMATCH,
-				  tt_uuid_str(replicaset_uuid),
-				  tt_uuid_str(&REPLICASET_UUID));
+
+	struct replica *master;
+
+	/*
+	 * Rebootstrap
+	 *
+	 * 1) Try to connect to all nodes in the replicaset during
+	 *    the waiting period in order to update their ballot.
+	 *
+	 * 2) After updated the ballot of all nodes, it are looking
+	 *    for a new master.
+	 *
+	 * 3) Wait until bootstrap_from_master succeeds.
+	 */
+	while (true) {
```

Comment:
There's a good comment in the issue description:
> Retry itself should re-create the appliers to fetch new ballots, try to find a new join-master, and retry. Retry would need to be done via polling with small sleeps if a leader can not be found. So as this wouldn't be a busy loop.

Let's do that. I missed that completely.

Reply:
Done. I'm not sure what's right

Reply:
I should've pasted a smaller quote, sorry.
I meant you should add small sleeps between loop iterations so that it's not a busy loop.
Vlad already mentioned it in his comment.